net/transport_layer/udp.h

```diff
@@ -0,0 +1,50 @@
+#ifndef LAUNCH_CORE_UDP_H
+#define LAUNCH_CORE_UDP_H
+
+#include "../socket/Socket.h"
+#include "../packet/Packet.h"
+#include "../network_layer/NetworkLayer.h"
+
+typedef struct {
+    sockaddr_t src,
+    sockaddr_t dst,
+    uint32_t length,
+    int checksum,
+    uint8_t *data_octets
+} UDP_HEADER_T;
+
+class UDP : public NetworkLayer {
```

Comment:
maybe call this UDPLayer or UDPRouter, it's a little more descriptive and matches what the other network layer classes are called